corehq/ex-submodules/pillowtop/pillow/interface.py

```diff
@@ -114,24 +114,22 @@ def _should_process_in_chunks(self):
             and getattr(self.processors[0], 'processor_chunk_size', 0) > 0)
 
     def process_changes(self, since, forever):
-        """
-        Handle changes stream from the changes stream and pass it to processors
-        """
+        # Pass changes from the feed to processors
         context = PillowRuntimeContext(changes_seen=0)
         try:
             changes_chunk = []
             for change in self.get_change_feed().iter_changes(since=since or None, forever=forever):
-                if change:
```

Comment:
Mainly because we need to skip `self._update_checkpoint(None, None)` if it's a empty change and if it's the processor is bulk. Also, it simplifies this block

Reply:
Actually, reverting that change